numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -242,17 +242,36 @@ gentype_nonzero_number(PyObject *m1)
 static PyObject *
 genint_type_str(PyObject *self)
 {
-    PyObject  *item, *item_str;
+    PyObject  *item, *item_str, *dtype, *dtype_name;
     item = gentype_generic_method(self, NULL, NULL, "item");
     if (item == NULL) {
         return NULL;
     }
 
-    item_str = PyObject_Str(item);
+    if ((dtype = PyObject_GetAttrString(self, "dtype")) == NULL) {
+        return NULL;
+    }
+
+    if ((dtype_name = PyObject_GetAttrString(dtype, "name")) == NULL) {
+        return NULL;
+    }
+
+    item_str = PyUString_FromFormat("numpy.%S(%S)", dtype_name, item);
```

Comment:
This feels like overkill - does the following work?
```suggestion
    item_str = PyUString_FromFormat("%s(%S)", Py_TYPE(self)->tp_name, item);
```

Reply:
This is much better :smile:  . Didn't think of this, thanks.

Reply:
@eric-wieser , so just a doubt regarding debugging, how do you gdb into that code? currently, I put raise`(SIGINT)` and do `gdb python`, then do `run code.py` that hits that. Is there a better way?

Reply:
I haven't had to jump into GDB in a while, so I'm afraid I don't remember my usual trick. Are you asking because this is giving the wrong result?

Reply:
Oh I see, I was just curious about how numpy folks debug in general, some uncommon techniques/tools, etc to simplify stuff than just raw gdb.